api/api.go

```diff
@@ -2,22 +2,80 @@ package api
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
 
-	"github.com/gorilla/mux"
+	"github.com/ONSdigital/dp-feedback-api/config"
+	"github.com/ONSdigital/dp-feedback-api/middleware"
+	"github.com/ONSdigital/log.go/v2/log"
+	"github.com/go-chi/chi/v5"
 )
 
 // API provides a struct to wrap the api around
 type API struct {
-	Router *mux.Router
+	Cfg         *config.Config
+	Router      chi.Router
+	EmailSender EmailSender
 }
 
 // Setup function sets up the api and returns an api
-func Setup(ctx context.Context, r *mux.Router) *API {
+func Setup(ctx context.Context, cfg *config.Config, r chi.Router, e EmailSender) *API {
 	api := &API{
-		Router: r,
+		Cfg:         cfg,
+		Router:      r,
+		EmailSender: e,
 	}
 
-	// TODO: remove hello world example handler route
-	r.HandleFunc("/hello", HelloHandler(ctx)).Methods("GET")
+	api.mountEndpoints(ctx)
+
 	return api
 }
+
+// mountEndpoints creates a a new chi Router with the auth middleware and required endpoints,
+// and then mounts it to the existing router, in order to prevent existing endpoints (i.e. /health) to go through auth.
+func (api *API) mountEndpoints(ctx context.Context) {
+	r := chi.NewRouter()
+	middleware.Setup(r, api.Cfg.ServiceAuthToken)
+
+	// TODO: remove hello world example handler route when component tests are no longer using it
+	r.Get("/hello", HelloHandler(ctx))
```

Comment:
And the HelloHandler?

Reply:
Yes, I left it because it would break the component tests, so I decided to do this in the component tests task:
https://github.com/ONSdigital/dp-feedback-api/pull/4
Is it ok if we leave it for this pr? (it will be removed when we merge the next one anyway)